From 574bfdb4b429d8b80336a0749e41840f34b4e2fc Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Sun, 23 Jun 2024 21:51:56 +0200
Subject: [PATCH 36/41] ARM: dts: stm32: Add ethernet support for DH
 STM32MP13xx DHCOR DHSBC board

Add ethernet support for the DH STM32MP13xx DHCOR DHSBC carrier board.
This carrier board is populated with two gigabit ethernet ports and two
Realtek RTL8211F PHYs, both are described in this DT patch.

Upstream-Status: Backport [1b02383c385b16b4b275e30a3dd5860c0fd95c4a]
Signed-off-by: Marek Vasut <marex@denx.de>
Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
---
 .../boot/dts/st/stm32mp135f-dhcor-dhsbc.dts   | 56 +++++++++++++++++++
 1 file changed, 56 insertions(+)

diff --git a/arch/arm/boot/dts/st/stm32mp135f-dhcor-dhsbc.dts b/arch/arm/boot/dts/st/stm32mp135f-dhcor-dhsbc.dts
index 5f4f6b6e427a5..bacb70b4256bc 100644
--- a/arch/arm/boot/dts/st/stm32mp135f-dhcor-dhsbc.dts
+++ b/arch/arm/boot/dts/st/stm32mp135f-dhcor-dhsbc.dts
@@ -22,6 +22,8 @@ / {
 		     "st,stm32mp135";
 
 	aliases {
+		ethernet0 = &ethernet1;
+		ethernet1 = &ethernet2;
 		serial2 = &usart1;
 //		serial3 = &usart2;
 	};
@@ -72,6 +74,60 @@ channel@12 {
 	};
 };
 
+&ethernet1 {
+	phy-handle = <&ethphy1>;
+	phy-mode = "rmii";
+	pinctrl-0 = <&eth1_rgmii_pins_a>;
+	pinctrl-1 = <&eth1_rgmii_sleep_pins_a>;
+	pinctrl-names = "default", "sleep";
+ max-speed = <100>;
+	status = "okay";
+
+	mdio {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "snps,dwmac-mdio";
+
+		ethphy1: ethernet-phy@1 {
+			/* RTL8211F */
+//AAA			compatible = "ethernet-phy-id001c.c916";
+//AAA			interrupt-parent = <&gpiog>;
+//AAA			interrupts = <12 IRQ_TYPE_LEVEL_LOW>;
+			reg = <0>;
+			reset-assert-us = <15000>;
+			reset-deassert-us = <55000>;
+			reset-gpios = <&gpioa 11 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+//AAA&ethernet2 {
+//AAA	phy-handle = <&ethphy2>;
+//AAA	phy-mode = "rgmii-id";
+//AAA	pinctrl-0 = <&eth2_rgmii_pins_a>;
+//AAA	pinctrl-1 = <&eth2_rgmii_sleep_pins_a>;
+//AAA	pinctrl-names = "default", "sleep";
+//AAA	st,ext-phyclk;
+//AAA	status = "okay";
+//AAA
+//AAA	mdio {
+//AAA		#address-cells = <1>;
+//AAA		#size-cells = <0>;
+//AAA		compatible = "snps,dwmac-mdio";
+//AAA
+//AAA		ethphy2: ethernet-phy@1 {
+//AAA			/* RTL8211F */
+//AAA			compatible = "ethernet-phy-id001c.c916";
+//AAA			interrupt-parent = <&gpiog>;
+//AAA			interrupts = <15 IRQ_TYPE_LEVEL_LOW>;
+//AAA			reg = <1>;
+//AAA			reset-assert-us = <15000>;
+//AAA			reset-deassert-us = <55000>;
+//AAA			reset-gpios = <&gpiog 8 GPIO_ACTIVE_LOW>;
+//AAA		};
+//AAA	};
+//AAA};
+
 &gpioa {
 	gpio-line-names = "", "", "", "",
 			  "", "DHSBC_USB_PWR_CC1", "", "",
-- 
2.43.0

